### PR TITLE
Minor fix for libc/pwd

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- libc/pwd: Reuse g_passwd and g_passwd_buffer in getpwbuf
- libc/pwd: Correct the geos related comment and ddd ROOT_GEOS macro

## Impact

code refactor

## Testing

ci

### CODE REVIEW

File: libs/libc/pwd/lib_find_pwdfile.c [hunk 1]
```diff
@@ -133,13 +133,14 @@ static int pwd_foreach(pwd_foreach_match_t match, uintptr_t arg,
...
    *
-   *   user:x:uid:uid:home
+   *   user:x:uid:uid:geos:home
    *
    * Where:
    *   user:  User name
    *   x:     Encrypted password
    *   uid:   User ID
    *   uid:   Group ID
+   *   geos： User information
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply libs/libc/pwd/lib_find_pwdfile.c:143]: But why you suggest to not align with other line?

**Review by pkarashchenko — APPROVED** [review]

File: libs/libc/pwd/lib_find_pwdfile.c [hunk 2]
```diff
@@ -133,13 +133,14 @@ static int pwd_foreach(pwd_foreach_match_t match, uintptr_t arg,
...
    *   home:  Login directory
```
Comment (pkarashchenko) [inline libs/libc/pwd/lib_find_pwdfile.c:144]:
```suggestion
   *   uid:   Group ID
   *   geos:  User information
   *   home:  Login directory
```

File: libs/libc/pwd/lib_find_pwdfile.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (patacongo) [reply libs/libc/pwd/lib_find_pwdfile.c:143]: There appears to be a TAB character after "geos:"

File: libs/libc/pwd/lib_find_pwdfile.c (hunk 2 above)
Comment (xiaoxiang781216) [reply libs/libc/pwd/lib_find_pwdfile.c]: Ok, I found the reason, I use the wrong `:`.

**Pull request merged**